.github/workflows/publish.yaml

```diff
@@ -37,19 +35,36 @@ jobs:
     steps:
       - uses: actions/setup-node@v1
         with:
-          node-version: 15
+          node-version: 16
           registry-url: 'https://registry.npmjs.org'
+
       - name: Checkout
         uses: actions/checkout@v2
-        with:
-          fetch-depth: 1
+
+      - name: Setup env. variables
+        run: |
+        if [[ "${{ github.event_name }}" == "repository_dispatch" && "${{ github.event.action }}"== "build-images" ]]; then
+          echo "BEE_VERSION=${{ github.event.client_payload.tag }}" >> $GITHUB_ENV
+          echo "BUILD_IMAGE=true" >> $GITHUB_ENV
+          echo "COMMIT_VERSION_TAG=true" >> $GITHUB_ENV
+          echo "BUILD_IMAGE=false" >> $GITHUB_ENV
+          echo "STATE_COMMIT=true" >> $GITHUB_ENV
+        else
+          echo "BEE_VERSION=${{ github.event.inputs.beeVersion }}" >> $GITHUB_ENV
+          echo "BUILD_IMAGE=${{ github.event.inputs.beeVersionAsCommitHash }}" >> $GITHUB_ENV
+          echo "COMMIT_VERSION_TAG=${{ github.event.inputs.commitVersionTag }}" >> $GITHUB_ENV
+          echo "BUILD_IMAGE=${{ github.event.inputs.beeVersionAsCommitHash }}" >> $GITHUB_ENV
+          echo "STATE_COMMIT=${{ github.event.inputs.stateCommit }}" >> $GITHUB_ENV
+        fi
```

Comment:
wouldn't it be better if you had initialized the github environments under the `end` as you do here at the first condition block and otherwise override it if there is `github.event.inputs`? Maybe it would seem cleaner.

Reply:
Good idea, let me know if the patch addresses it as you imagined.